src/stat_tests.py

```diff
@@ -45,7 +40,7 @@ def main(data_path, output_file):
     results = {}
     for i, file in enumerate(files):
         # analysis\preprocessing\*.csv
-        df = pd.read_csv(file)
+        df = pd.read_csv(data_path + '/' + file)
```

Comment:
When I run the stat_test.py script I get this error (screenshot) - I remember Maeve and I fixing this last week to remove the `data_path+ '/'` to get it working.. 
I think it might be the `preprocessed_data_files` which is already concatenating the `data_path` in the glob so appends it twice when it gets to line 43.
Lmk if you get something different, might be a different reason for this error.

<img width="1121" alt="Screenshot 2021-12-03 at 10 43 29" src="https://user-images.githubusercontent.com/40150248/144656030-be312dee-3d02-42bb-bceb-6a6f8d242043.png">

Reply:
YEah I had to add it for it to work on my local. Tested end-to-end and it worked for me.

Reply:
Here is a successful run of the pipeline
![image](https://user-images.githubusercontent.com/75291170/144685582-0178d9db-2938-482c-83cb-44da01e7a33b.png)

I'm beginning to wonder if there is an OS-specific difference in how `glob` behaves..

Reply:
printing out the `files` variable in stats-test returns the following (as expected)

['addiction_processed.csv', 'adhd_processed.csv', 'alcoholism_processed.csv', 'anxiety_processed.csv', 'autism_processed.csv', 'bipolarreddit_processed.csv', 'bpd_processed.csv', 'depression_processed.csv', 'EDAnonymous_processed.csv', 'healthanxiety_processed.csv', 'lonely_processed.csv', 'ptsd_processed.csv', 'schizophrenia_processed.csv', 'socialanxiety_processed.csv', 'suicidewatch_processed.csv']

Can you see what it's outputting on your local? Print out data-path as well.

Reply:
Ya, this is my output:

```
data path:  data/processed/
printing files:  ['data/processed/bipolarreddit_processed.csv', 'data/processed/EDAnonymous_processed.csv', 'data/processed/socialanxiety_processed.csv', 'data/processed/alcoholism_processed.csv', 'data/processed/lonely_processed.csv', 'data/processed/healthanxiety_processed.csv', 'data/processed/ptsd_processed.csv', 'data/processed/suicidewatch_processed.csv', 'data/processed/addiction_processed.csv', 'data/processed/bpd_processed.csv', 'data/processed/autism_processed.csv', 'data/processed/schizophrenia_processed.csv', 'data/processed/adhd_processed.csv', 'data/processed/depression_processed.csv', 'data/processed/anxiety_processed.csv']
```

When printing out `glob.glob(data_path + '/*.csv')` I also get:
```
['data/processed/bipolarreddit_processed.csv', 'data/processed/EDAnonymous_processed.csv', 'data/processed/socialanxiety_processed.csv', 'data/processed/alcoholism_processed.csv', 'data/processed/lonely_processed.csv', 'data/processed/healthanxiety_processed.csv', 'data/processed/ptsd_processed.csv', 'data/processed/suicidewatch_processed.csv', 'data/processed/addiction_processed.csv', 'data/processed/bpd_processed.csv', 'data/processed/autism_processed.csv', 'data/processed/schizophrenia_processed.csv', 'data/processed/adhd_processed.csv', 'data/processed/depression_processed.csv', 'data/processed/anxiety_processed.csv']
```
so it might be that it's not splitting correctly?

Reply:
I bet that's exactly what's happening; different OS might handle the following component of
`preprocessed_data_files = [datafile.split('\\')[-1] for datafile in glob.glob(data_path + '/*.csv')] # '..\..\data\*.csv' ` differently....

`datafile.split('\\')[-1]`

Reply:
my glob.glob(data_path + '/*.csv) returns:

['data/processed\\addiction_processed.csv', 'data/processed\\adhd_processed.csv', 'data/processed\\alcoholism_processed.csv', 'data/processed\\anxiety_processed.csv', 'data/processed\\autism_processed.csv', 'data/processed\\bipolarreddit_processed.csv', 'data/processed\\bpd_processed.csv', 'data/processed\\depression_processed.csv', 'data/processed\\EDAnonymous_processed.csv', 'data/processed\\healthanxiety_processed.csv', 'data/processed\\lonely_processed.csv', 'data/processed\\ptsd_processed.csv', 'data/processed\\schizophrenia_processed.csv', 'data/processed\\socialanxiety_processed.csv', 'data/processed\\suicidewatch_processed.csv']

Reply:
Yeah I think you're right about it being OS-specific (scroll down to String Manipulation):
https://www.earthdatascience.org/courses/intro-to-earth-data-science/python-code-fundamentals/work-with-files-directories-paths-in-python/os-glob-manipulate-file-paths/


<img width="821" alt="Screenshot 2021-12-03 at 15 45 24" src="https://user-images.githubusercontent.com/40150248/144686706-dc9bfdeb-1a34-4792-81cb-f05966572697.png">

Reply:
So...maybe we say our project is only supported on one or the other?? Just kidding ahah...but this is a nuisance. I'll try refactoring to use `data_folder.split(os.sep)` and see if I can get this working on my local then will push the changes. Stand by

Reply:
@mel-liow Changes are up - had to refactor R datadownload script, stat_tests.py, and eda_script.py

Reply:
Speedy! Will pull changes and check locally

Reply:
Ya all good!